### PR TITLE
War Rations fix

### LINKED ISSUE #36: War Rations doesn't work
Doesn't reduce amount of food consumed by 20%

### PR DESCRIPTION
**Pull request by Tyler-IN**
Carrying on where @Skau left off tonight;

before

after

Fixes #36.

### CODE REVIEW

**Comment by Tyler-IN** [comment]
In case someone (probably TaleWorlds) changes the perk, it'll still take affect according to the expected behavior.

**Comment by Tyler-IN** [comment]
There's a separate branch for options that does handles like HugsLib, e.g. Option<T>

File: src/CommunityPatch/Patches/WarRationsPerkPatch.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        // qm is not in party
+        && party.MemberRoster.All(element => element.Character?.HeroObject != qm)
+      )
+        return;
+
+      var perk = ActivePatch._perk;
+      if (!qm.GetPerkValue(perk))
+        return;
+
+      var explainedNumber = new ExplainedNumber(__result, explainer);
```
Comment (tynakuh) [inline src/CommunityPatch/Patches/WarRationsPerkPatch.cs:78]: I have a feeling this is going to be a common thing we'll have to do, wonder if we should build a helper function

Comment (Tyler-IN) [reply src/CommunityPatch/Patches/WarRationsPerkPatch.cs:78]: Possibly. Copy pasta for now, refactor to helper later.

**Pull request merged**